engine/render/src/render/render.cpp

```diff
@@ -628,61 +631,68 @@ namespace dmRender
         for (uint32_t i = 0; i < render_context->m_RenderObjects.Size(); ++i)
         {
             RenderObject* ro = render_context->m_RenderObjects[i];
+            if (ro->m_VertexCount == 0)
+                continue;
+
+            MaterialTagList taglist;
+            uint32_t taglistkey = dmRender::GetMaterialTagListKey(ro->m_Material);
+            dmRender::GetMaterialTagList(render_context, taglistkey, &taglist);
 
-            if (ro->m_VertexCount > 0 && (GetMaterialTagMask(ro->m_Material) & tag_mask) == tag_mask)
+            if (predicate && !dmRender::MatchMaterialTags(taglist.m_Count, taglist.m_Tags, predicate->m_TagCount, predicate->m_Tags))
```

Comment:
Doing this per-renderobject is more costly, but we also only use this function in a test app.